src/components/PageHeadingArtifact.vue

```diff
@@ -32,7 +31,7 @@
   const crumbs = computed<BreadCrumbs>(() => {
```

Comment:
I was trying to understand the logic here and rewrote it this way. Feel free to do nothing with this

```typescript
  const crumbs = computed<BreadCrumbs>(() => {
    const artifactIdCrumb: BreadCrumb = {
      text: props.artifact.id
    }

    if(props.artifact.key) {
      return [
        { 
          text: props.artifact.key, 
          to: routes.artifactKey(props.artifact.id) 
        },
        artifactIdCrumb
      ]
    }

    const internalCrumbs: BreadCrumbs = []

    if(flowRun.value) {
      internalCrumbs.push({
        text: flowRun.value.name ?? '',
        to: routes.flowRun(flowRun.value.id),
      })
    }

    if (taskRun.value) {
      internalCrumbs.push({
        text: taskRun.value.name ?? '',
        to: routes.taskRun(taskRun.value.id),
      })
    }

    internalCrumbs.push(artifactIdCrumb)

    return internalCrumbs
  })
```